addons/website_blog/static/src/js/systray_items/new_content.js

```diff
@@ -6,8 +6,26 @@ patch(NewContentModal.prototype, {
         super.setup();
 
         const newBlogElement = this.state.newContentElements.find(element => element.moduleXmlId === 'base.module_website_blog');
-        newBlogElement.createNewContent = () => this.onAddContent('website_blog.blog_post_action_add', true);
+        newBlogElement.createNewContent = () => this.onAddContent("website_blog.blog_post_action_add", true, this.getCurrentBlogContext());
         newBlogElement.status = MODULE_STATUS.INSTALLED;
         newBlogElement.model = 'blog.post';
     },
+
+    getCurrentBlogContext() {
+        // Using iframe to access mainObject to check if we are on blog page
+        const iframe = document.querySelector("iframe.o_iframe");
+        const isBlogPage = iframe.contentDocument.documentElement.dataset.mainObject?.startsWith("blog");
+
+        if (isBlogPage) {
+            const blogEl = iframe.contentDocument.querySelector(
+                "#wrap.website_blog [data-oe-model='blog.blog']"
+            );
```

Comment:
Let's go with that then...
```suggestion
        const iframeEl = document.querySelector("iframe.o_iframe").contentDocument;
        const isBlogPage = iframeEl.documentElement.dataset.mainObject?.startsWith("blog");

        if (isBlogPage) {
            const blogEl = iframeEl.querySelector(
                "#wrap.website_blog [data-oe-model='blog.blog']"
            );
```